rfcs/2021-11-xx-9862-ingest-apm-stats-along-traces-in-dd-agent-source.md

```diff
@@ -0,0 +1,164 @@
+[PR][sketch-pr]). In the proto definition sketches are stored as unstructured bytes slices, but those
+[fields][call-to-toproto] are filled with a [protobuf encoded ddsketch][ddsktech-proto]. Given the sketch implementation
+in Vector is also [heavily based on ddsketch][vector-sketch]  this should not cause any significant accuracy loss
+(**TBC**).
+
+This raises the question of having a second (assuming that the `datadog-agent` sources accepts Datadog Agent metrics -
+[RFC][dd-agent-metric-rfc]), unrelated, metric stream coming out of the `datadog-agent` source. Obviously metrics
+ingested from APM stats will have to be routed along with traces, and most often they will follow a different path that
+other plain metrics. Thus it is suggested to re-arrange the `datadog-agent` source according to the following points
+(Additional details on what exactly are "Datadog Agents" can be found in the trace support [RFC][trace-support-pr] and
+may provide relevant context for undermentionned points):
+
+* Keep one Vector source per kind of Datadog Agent
+* Then we would have the following Vector sources:
+  * `datadog-core`: would be the current `datadog-agent` sources, it would receive the data sent by the Datadog "core" Agent (the agent that collects logs & metrics). Note that renaming this source is probably optional, but documentation update would be required anyway.
+  * `datadog-trace`: would support all data sent by the `trace-agent`
+  * And so on as the support list grow:
+    * `datadog-process` source for the Datadog `process-agent`
+    * `datadog-security` for the Datadog security Agent
+    * Etc.
+
+[apm-stats-proto]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/pb/stats.proto
+[client-side-marker]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/pb/stats.proto#L15
+[concentrator]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/stats/concentrator.go
+[client-stats-aggregator]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/stats/client_stats_aggregator.go#L25-L45
+[stats-writer]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/writer/stats.go
+[client-grouped-stats-proto]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/pb/stats.proto#L55-L70
+[sketch-pr]: https://github.com/vectordotdev/vector/pull/9178
+[call-to-toproto]: https://github.com/DataDog/datadog-agent/blob/dc2f202/pkg/trace/stats/statsraw.go#L53-L62
+[ddsktech-proto]: https://github.com/DataDog/sketches-go/blob/0829b5a6a3c9627d1faee6c690611067ee110e38/ddsketch/pb/ddsketch.proto
+[vector-sketch]: https://github.com/vectordotdev/vector/blob/4db23b3/lib/vector-core/src/metrics/ddsketch.rs#L202-L213
+[dd-agent-metric-rfc]: https://github.com/vectordotdev/vector/pull/9057
+
+## Cross cutting concerns
+
+None identified so far.
+
+## Scope
+
+### In scope
+
+- Decode APM request (protobuf) received from the trace agent
+- Convert those into a Vector metric internal representation
```

Comment:
The description up to makes it sound like we cannot separately send metric data that was extracted from an APM trace, that it must be sent as part of a trace, not standalone as a normal metric.

Is that not true?

Reply:
No, apm stats are sent to the same host by the `trace-agent` but on different path (`/api/v0.2/stats` & `/api/v0.2/traces`)  so they are flowing separately and as long as identifiers remain intact they can be dealt with separately and correlation will happen in the backend.

Reply:
We could do it but metrics sent to the "normal" metrics intake are not accounted by the APM product. They need to be sent to their own endpoint (https://trace.agent.datadoghq.com/api/v0.2/stats). Given the fact that we will probably, in a not-so-far-future, compute APM stats for 3rd party traces, I think this should happen in a `datadog-traces` sink, in that extend I think that pre-cooked APM stats should probably be sent by this very same sink.